### PR TITLE
When To Retire.

### LINKED ISSUE #3: Retirement
Input user's current age, annual salary, percentage saved (savings matched by employer - so double amount saved). Input desired retirement savings goal. Output what age savings goal will be met. You can assume death at 100 years (therefore, indicate if the savings goal is not met).

### PR DESCRIPTION
**Pull request by EliCDavis**
Created a function that would determine when a person could retire.  Some boundaries could be vague at what to do at them and might require further clarification.  As of now it does a lot of parameter checking.

Closes #3

### CODE REVIEW

**Comment by EliCDavis** [comment]
Oh one line doesn't get hit huh.  One second

File: Assignment1/retirement.py [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+        raise ValueError("Saving must be a number!")
+
+    if isinstance(goal, (int, float)) is False:
+        raise ValueError("Goal must be a number!")
+
+    if age < 0:
+        raise ValueError("Can not have a negative age")
+
+    if salary <= 0:
+        raise ValueError("Can not have a positive salary")
```
Comment (JacobAMason) [inline Assignment1/retirement.py]: I think this should say "Can not have a negative salary" or "Salary must be greater than 0"

File: Assignment1/retirement.py [hunk 2]
```diff
@@ -1,10 +1,38 @@
-class Retirement:
-    def __init__(self):
-        pass
+def year_can_retire(age, salary, saving, goal):
```
Comment (JacobAMason) [inline Assignment1/retirement.py]: Might want to rename "saving" to "savingPercent" or something that makes it clear that this value is a percentage. Even though I knew it was a percentage, my brain kept wanting to swap in "savings" instead.

**Review by JacobAMason — CHANGES_REQUESTED** [review]
I can think of another test: if the goal is 0, the salary and savings could be 0 and the goal would be met at the current age. It might be more than it's worth for this one exception, though.

**Comment by EliCDavis** [comment]
@JacobAMason I've taken into account your suggestions and then created test classes to break up everything.

**Review by JacobAMason — APPROVED** [review]

**Pull request merged**